### DISCUSSION TITLE
Different permissions based on manifest version?

### DISCUSSION THREAD

**Discussion by max-arias** [wxt-dev/wxt#172] [Q&A] [answered]
ie.

For MV3, I need: `['storage', 'scripting', 'nativeMessaging'],`
For MV2, I need: `['storage', '*://*/*', '<all_urls>', 'nativeMessaging'],`

I tried:

```
 manifest: {
    permissions: {
      mv2: ...
      mv2: ...
```

**Comment by aklinker1** [comment, marked as answer]
The manifest option only accepts actual manifest schemas, so having a `mv2` and `mv3` fields won't do anything.

If you need to customize the manifest per build target, you should set it equal to a function to return the manifest depending on the manifest version of the build:

```ts
// wxt.config.ts
export default defineConfig({
  manifest: ({ manifestVersion }) => {
    const permissions = ["storage", "nativeMessaging"];
    if (manifestVersion === 2) permissions.push("*://*/*", "<all_urls>");
    else permissions.push("scripting");
    
    return {
      permissions,
    }, 
  },
}
```

I was going to link the [config reference](https://wxt.dev/api/config.html#manifest), but the type listed there is wrong... I'll get that fixed. Created https://github.com/wxt-dev/wxt/issues/173 for updating it.

**Reply by maxarias-io** [reply]
Thanks 👍
